### PR TITLE
Bash sanitization

### PR DESCRIPTION
**Pull request by rhcarvalho**
Tested deploy and scale up scenario described in #125.

This should give us clearer errors in the logs instead of failure to connect to "Thu" (because today is Thursday, right?) like @php-coder was seeing :scream_cat:

### CODE REVIEW

**Comment by rhcarvalho** [comment]
@bparees ptal

File: 2.4/root/usr/share/container-scripts/mongodb/common.sh [hunk 1]
```diff
@@ -37,8 +41,10 @@ function cache_container_addr() {
 
 # wait_for_mongo_up waits until the mongo server accepts incomming connections
 function wait_for_mongo_up() {
-  local mongo_host="${1-}"
-  local mongo_cmd="mongo admin "
+  local mongo_host
+  mongo_host="${1-}"
```
Comment (php-coder) [inline 2.4/root/usr/share/container-scripts/mongodb/common.sh:44]: What is the benefit of using 2 lines for variable's initialization?

Comment (rhcarvalho) [inline 2.4/root/usr/share/container-scripts/mongodb/common.sh:44]: It is explained here: https://google.github.io/styleguide/shell.xml?showone=Use_Local_Variables#Use_Local_Variables

**Comment by bparees** [comment]
lgtm and thanks for the education on the use of local.  Seems to have a test issue though.

**Comment by bparees** [comment]
[test]

**Comment by rhcarvalho** [comment]
The test result says "Yum repo mirrorlist timed out"

Let's try again [test]

**Comment by bparees** [comment]
@rhcarvalho that test result is a red herring, it's not the actual failure cause.

**Comment by bparees** [comment]
(i'm pretty sure we always get that error now, even when things succeed.  if you have time to look into which repo is timing out and see if we can remove it from our docker image or something, that'd be great)

**Comment by rhcarvalho** [comment]
Aha, so there's a test failure that reads:

```
MongoDB shell version: 2.4.9
connecting to: 172.17.0.2:27017/db
Thu Feb 25 13:01:55.818 Error: couldn't connect to server 172.17.0.2:27017 at src/mongo/shell/mongo.js:147
    Trying to connect...
MongoDB shell version: 2.4.9
connecting to: 172.17.0.2:27017/db
Thu Feb 25 13:01:58.938 Error: couldn't connect to server 172.17.0.2:27017 at src/mongo/shell/mongo.js:147
  Giving up: Failed to connect. Logs:
=> Waiting for container IP address ...
Stopping and removing container 58ccaa388e21715596a6418d761f083fe531a43ef1716d48141e94087ac04224...
58ccaa388e21715596a6418d761f083fe531a43ef1716d48141e94087ac04224
58ccaa388e21715596a6418d761f083fe531a43ef1716d48141e94087ac04224
Done.
ERROR: mongodb-rhel7 failed testing.
```

Looks like a flake. Let's see what the next run gives (probably tomorrow for me).

The yum error goes with this output:

```
Step 6 : RUN yum install -y yum-utils &&     yum-config-manager --enable rhel-server-rhscl-7-rpms &&     yum-config-manager --enable rhel-7-server-optional-rpms &&     INSTALL_PKGS="bind-utils gettext iproute rsync tar v8314 mongodb24-mongodb mongodb24" &&     yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS &&     rpm -V $INSTALL_PKGS &&     yum clean all &&     mkdir -p /var/lib/mongodb/data && chown -R mongodb.0 /var/lib/mongodb/ &&     chmod -R a+rwx /opt/rh/mongodb24/root/var/lib/mongodb &&     chmod -R g+rwx /var/lib/mongodb
 ---> Running in f6f971e34169
Loaded plugins: ovl
Repo rhui-REGION-client-config-server-7 forced skip_if_unavailable=True due to: /etc/pki/rhui/cdn.redhat.com-chain.crt
Repo rhui-REGION-client-config-server-7 forced skip_if_unavailable=True due to: /etc/pki/rhui/product/rhui-client-config-server-7.crt
Repo rhui-REGION-client-config-server-7 forced skip_if_unavailable=True due to: /etc/pki/rhui/rhui-client-config-server-7.key
Repo rhui-REGION-rhel-server-releases forced skip_if_unavailable=True due to: /etc/pki/rhui/cdn.redhat.com-chain.crt
Repo rhui-REGION-rhel-server-releases forced skip_if_unavailable=True due to: /etc/pki/rhui/product/content-rhel7.crt
Repo rhui-REGION-rhel-server-releases forced skip_if_unavailable=True due to: /etc/pki/rhui/content-rhel7.key
Repo rhui-REGION-rhel-server-rh-common forced skip_if_unavailable=True due to: /etc/pki/rhui/cdn.redhat.com-chain.crt
Repo rhui-REGION-rhel-server-rh-common forced skip_if_unavailable=True due to: /etc/pki/rhui/product/content-rhel7.crt
Repo rhui-REGION-rhel-server-rh-common forced skip_if_unavailable=True due to: /etc/pki/rhui/content-rhel7.key
Could not retrieve mirrorlist https://rhui2-cds01.REGION.aws.ce.redhat.com/pulp/mirror/rhui-client-config/rhel/server/7/x86_64/os error was
14: curl#6 - "Could not resolve host: rhui2-cds01.REGION.aws.ce.redhat.com; Name or service not known"
Could not retrieve mirrorlist https://rhui2-cds01.REGION.aws.ce.redhat.com/pulp/mirror/content/dist/rhel/rhui/server/7/7Server/x86_64/os error was
14: curl#6 - "Could not resolve host: rhui2-cds01.REGION.aws.ce.redhat.com; Name or service not known"
Could not retrieve mirrorlist https://rhui2-cds01.REGION.aws.ce.redhat.com/pulp/mirror/content/dist/rhel/rhui/server/7/7Server/x86_64/rh-common/os error was
14: curl#6 - "Could not resolve host: rhui2-cds01.REGION.aws.ce.redhat.com; Name or service not known"
Package yum-utils-1.1.31-34.el7.noarch already installed and latest version
Nothing to do
```

**Comment by rhcarvalho** [comment]
[test]

**Comment by rhcarvalho** [comment]
@legionus please help! :)

**Comment by rhcarvalho** [comment]
[test]

**Comment by rhcarvalho** [comment]
@legionus you are the Bash guru! :clap: big thanks for your help

@bparees ^^ finally we got this green :)

**Comment by rhcarvalho** [comment]
@bparees @mfojtik ready to merge, ptal

**Comment by mfojtik** [comment]
can you squash pls?

**Comment by rhcarvalho** [comment]
@mfojtik there you go...

**Pull request merged**